src/adapters/EOFeedAdapter.sol

```diff
@@ -37,24 +38,29 @@ contract EOFeedAdapter is IEOFeedAdapter, Initializable {
      * @notice Initialize the contract
      * @param feedManager The feed manager address
      * @param feedId Feed id
-     * @param feedDecimals The decimals of the rate
+     * @param inputDecimals The decimals of the rate
+     * @param outputDecimals The decimals of the rate
      * @param feedDescription The description of feed
      * @param feedVersion The version of feed
      */
     function initialize(
         address feedManager,
         uint16 feedId,
-        uint8 feedDecimals,
+        uint8 inputDecimals,
+        uint8 outputDecimals,
         string memory feedDescription,
         uint256 feedVersion
     )
         external
         initializer
     {
         if (feedManager == address(0)) revert InvalidAddress();
+        if (inputDecimals == 0 || inputDecimals > 18) revert InvalidDecimals();
+        if (outputDecimals == 0 || outputDecimals > 18) revert InvalidDecimals();
         _feedManager = IEOFeedManager(feedManager);
         _feedId = feedId;
-        _decimals = feedDecimals;
+        _inputDecimals = inputDecimals;
+        _outputDecimals = outputDecimals;
```

Comment:
to save cost,
calculate the difference here and store it

Reply:
great idea.